packages/authentication/src/provider.ts

```diff
@@ -0,0 +1,79 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import * as http from 'http';
+import {HttpErrors, inject, ParsedRequest} from '@loopback/core';
+import {Provider} from '@loopback/context';
+import {Strategy} from 'passport';
+import {StrategyAdapter} from './strategy-adapter';
+
+// interface definition of a function which accepts a request and returns an authenticated user
+export interface AuthenticateFn
+{
+  (request: ParsedRequest): Promise<UserProfile>;
+}
+
+// interface definition of a user profile
+export interface UserProfile
+{
+  id?: string;
+  role?: string;
```

Comment:
Roles are typically not part of the user profile. OpenID connect defines the user info as http://openid.net/specs/openid-connect-core-1_0.html#StandardClaims. I think we should keep the interface minimum.